packages/open-next/src/adapters/routing/matcher.ts

```diff
@@ -154,7 +153,7 @@ export function handleRewrites<T extends RewriteDefinition>(
     }
   }
 
-  const queryString = urlQueryString ? `?${urlQueryString}` : "";
+  const queryString = query ? `?${convertToQueryString(query)}` : "";
```

Comment:
The issue might be here, query here could be `{}`, this will evaluate to true.
This does not explain why `query.test.ts` fails though.
My bet is that it comes from here https://github.com/sst/open-next/blob/8bc075b9696857d60ffe520154b8395c3a730b00/packages/open-next/src/adapters/event-mapper.ts#L126.
You should probably try deploying and add a bunch of console.logs to try to understand why it fails